test/Tester/HeterogeneousSilosTests/UpgradeTests.cs

```diff
@@ -87,17 +87,18 @@ public UpgradeTests()
             await StartSiloV2();
 
             // New activation should be V2
-            for (var i = numberOfGrains; i < numberOfGrains * 2; i++)
+            for (var i = 0; i < numberOfGrains * 2; i++)
             {
                 var grain = GrainFactory.GetGrain<IVersionUpgradeTestGrain>(i);
-                Assert.Equal(2, await grain.GetVersion());
+                var expectedVersion = i < numberOfGrains ? 1 : 2;
+                Assert.Equal(expectedVersion, await grain.GetVersion());
             }
```

Comment:
Sorry, but just add 2 loops to make the assert blatantly clear. Also checkout the stale code comment preceding it